jsonschema-maven-plugin/src/test/java/com/github/victools/jsonschema/plugin/maven/SchemaGeneratorMojoTest.java

```diff
@@ -0,0 +1,94 @@
+import java.io.StringReader;
+
+public class SchemaGeneratorMojoTest extends AbstractMojoTestCase {
+
+    private PlexusConfiguration pluginConfiguration;
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void setUp()
+            throws Exception {
+        super.setUp();
+
+        String pom =
+                "<project>\n" +
+                        "    <build>\n" +
+                        "        <plugins>\n" +
+                        "            <plugin>\n" +
+                        "                <groupId>com.github.victools</groupId>\n" +
+                        "                <artifactId>jsonschema-maven-plugin</artifactId>\n" +
+                        "                <version>4.8.0</version>\n" +
+                        "                <configuration>\n" +
+                        "                    <className>" +
+                        "com.github.victools.jsonschema.plugin.maven.TestClass</className>\n" +
+                        "                    <schemaFileName>result.json</schemaFileName>\n" +
+                        "                    <schemaFilePath>target/generated-test-sources</schemaFilePath>\n" +
+                        "                    <options>\n" +
+                        "                        <option>DEFINITIONS_FOR_ALL_OBJECTS</option>\n" +
+                        "                    </options>\n" +
+                        "                    <modules>\n" +
+                        "                        <module>\n" +
+                        "                            <name>Jackson</name>\n" +
+                        "                            <options>\n" +
+                        "                                <option>FLATTENED_ENUMS_FROM_JSONVALUE</option>\n" +
+                        "                            </options>\n" +
+                        "                        </module>\n" +
+                        "                        <module>\n" +
+                        "                            <name>MyModule</name>\n" +
+                        "                            <className>" +
+                        "com.github.victools.jsonschema.plugin.maven.TestModule</className>\n" +
+                        "                        </module>\n" +
+                        "                    </modules>\n" +
+                        "                </configuration>\n" +
+                        "            </plugin>\n" +
+                        "        </plugins>\n" +
+                        "    </build>" +
+                        "</project>";
+
+        Xpp3Dom pomDom = Xpp3DomBuilder.build(new StringReader(pom));
+        pluginConfiguration = extractPluginConfiguration("jsonschema-maven-plugin", pomDom);
+    }
+
+    /**
+     * Unit test to test the basic working with a complete configuration
+     *
+     * @throws Exception In case something goes wrong
+     */
+    public void testBasicWorking() throws Exception {
```

Comment:
Shouldn't there be a `@Test` annotation here?

Reply:
A `@Test` annotation would be more modern. I will work on the unit test a little more. On my list are:
- JUnit 4 annotations
- More unit tests with better coverage of the functionality.

For now I propose to merge this branch.